src/EnergyPlus/DXCoils.cc

```diff
@@ -7040,9 +7041,9 @@ void InitDXCoil(EnergyPlusData &state, int const DXCoilNum) // number of the cur
         }
 
         if (thisDXCoil.DXCoilType_Num == HVAC::CoilDX_CoolingTwoStageWHumControl) {
-            int DehumidModeNum; // Loop index for 1,Number of enhanced dehumidification modes
+            int DehumidModeNum;   // Loop index for 1,Number of enhanced dehumidification modes
+            int CapacityStageNum; // Loop index for 1,Number of capacity stages
             for (DehumidModeNum = 0; DehumidModeNum <= thisDXCoil.NumDehumidModes; ++DehumidModeNum) {
-                int CapacityStageNum; // Loop index for 1,Number of capacity stages
                 for (CapacityStageNum = 1; CapacityStageNum <= thisDXCoil.NumCapacityStages; ++CapacityStageNum) {
```

Comment:
You can delete lines 7044 and 7045 and update line 7046 with `for (int DehumidModeNum` and line 7047 with `for (int CapacityStageNum`. Same for lines 14665 and 15083 below.